distributed/variable.py

```diff
@@ -113,8 +113,14 @@ def __init__(self, scheduler):
             else:
                 if old["type"] == "Future":
                     await self.release(old["value"], name)
-            del self.waiting_conditions[name]
-            del self.variables[name]
+            try:
+                del self.waiting_conditions[name]
+            except KeyError:
+                pass
+            try:
+                del self.variables[name]
+            except KeyError:
+                pass
```

Comment:
We sometimes also use the `with ignoring(KeyError)` convention (I think that this now has some analog in the stdlib).  No pressure though.  Using straight try/except is probably understandable by a broader set of readers.